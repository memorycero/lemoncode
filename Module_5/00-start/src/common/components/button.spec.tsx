import * as React from 'react';
import { render, fireEvent, getByLabelText } from '@testing-library/react';
import { Button } from './button';

describe('Button component specs', () => {
    it('should display a default button type with a label', () => {
        // Arrange
        const props = {
            label: 'Click Me'
        };

        // Act
        const { getByText, asFragment } = render(<Button {...props} />);

        // Assert
        getByText(props.label);
        expect(asFragment()).toMatchSnapshot();

    });

    it('should call onClick property when it clicks on button', () => {
        // Arrange
        const props = {
            label: 'Click Me',
            onClick: jest.fn(),      
        };

        // Act
        const { getByText } = render(<Button {...props} />);

        const buttonElement = getByText('Click Me');
        fireEvent.click(buttonElement);

        // Assert
        expect(props.onClick).toHaveBeenCalled();
        expect(props.onClick).toHaveBeenCalledTimes(1);

    });
})