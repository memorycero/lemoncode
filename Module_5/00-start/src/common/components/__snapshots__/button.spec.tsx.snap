// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component specs should display a default button type with a label 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label"
    >
      Click Me
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
